dandi/files/__init__.py

```diff
@@ -128,10 +131,17 @@ def find_dandi_files(
                     # we may add later), so traverse through it as a regular
                     # directory.
                     if (p / BIDS_DATASET_DESCRIPTION).exists():
-                        bids2 = dandi_file(p / BIDS_DATASET_DESCRIPTION, dandiset_path)
-                        assert isinstance(bids2, BIDSDatasetDescriptionAsset)
-                        bidsdd = bids2
-                    path_queue.extend((q, bidsdd) for q in p.iterdir())
+                        # No nested BIDS.
```

Comment:
Please explain this.  Are BIDS datasets not allowed to be contained within other BIDS datasets?  I thought they were.

Reply:
They can, but if they are they are what's known as “derivatives”. Those have different and overall far more permissive and less agreed on rules for validation. Might be worth considering whether we want to support these at some point, but currently the upstream validator doesn't support them, so we shouldn't try to either.

Reply:
I've pushed a commit (ccc8d79) that updates the code & tests to treat `dataset_description.json` files beneath other `dataset_description.json` files as generic BIDS assets that don't start new BIDS datasets.  Also, `find_bids_dataset_description()` now finds the topmost `dataset_description.json` file.

Reply:
Thank you :)